src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +46,1034 @@ const ActorInit En_Niw_InitVars = {
+        this->actor.flags &= ~0x1; // targetable OFF
+        this->unknownState28E = 4;
+        this->actionFunc = EnNiw_Held;
+        this->actor.speedXZ = 0.0f;
+    } else {
+        if (this->unkTimer252 > 5) {
+            func_800B8BB0(&this->actor, globalCtx);
+        }
+        func_80891320(this, globalCtx, 2);
+    }
+}
+
+// action func: swimming and flying away after swimming
+void EnNiw_Swimming(EnNiw* this, GlobalContext* globalCtx) {
+    Vec3f ripplePos;
+
+    // even if hitting water, keep calling for reinforcements
+    // this should just be in update
+    if (this->isStormActive) {
+        EnNiw_SpawnAttackNiw(this, globalCtx); // spawn attack niw
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80891D78.asm")
+    this->actor.speedXZ = 2.0f;
+    if (this->actor.bgCheckFlags & 0x20) {
+        // still touching water
+        this->actor.gravity = 0.0f;
+        if (this->actor.yDistToWater > 15.0f) {
+            this->actor.world.pos.y += 2.0f;
+        }
+        if (this->unkTimer250 == 0) {
+            this->unkTimer250 = 30;
+            Math_Vec3f_Copy(&ripplePos, &this->actor.world.pos);
+            ripplePos.y += this->actor.yDistToWater;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80891F60.asm")
+            EffectSsGRipple_Spawn(globalCtx, &ripplePos, 100, 500, 30);
+        }
+        if (this->actor.bgCheckFlags & 8) {
+            this->actor.velocity.y = 10.0f;
+            this->actor.speedXZ = 1.0f;
+        }
+    } else {
+        this->actor.gravity = -2.0f;
+        if (this->actor.bgCheckFlags & 8) {
+            // has hit a wall
+            this->actor.velocity.y = 10.0f; // to the moon
+            this->actor.speedXZ = 1.0f;
+            this->actor.gravity = 0.0f;
+        } else {
+            this->actor.speedXZ = 4.0f;
+        }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_808920A0.asm")
+        if (this->actor.bgCheckFlags & 1) {
+            this->actor.gravity = -2.0f;
+            this->unkTimer254 = 100;
+            this->unkTimer250 = 0;
+            this->actor.velocity.y = 0.0f;
+            if (!this->isStormActive) {
+                EnNiw_SetupRunning(this);
+            } else {
+                this->unknownState28E = 3;
+                this->actionFunc = EnNiw_CuccoStorm;
+            }
+        }
+    }
+    func_80891320(this, globalCtx, 2);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80892248.asm")
+void EnNiw_Trigger(EnNiw* this, GlobalContext* globalCtx) {
+    // ! @ BUG: I believe multiple values got optimized out
+    // value has to be a temp to match (v0) but there is only one value
+    // explained by there being code where value got assigned with multiple conditions
+    // suspicious: in vanilla there is only one cucco variant (0xFFFF)
+    s32 value;
+    if (1) {
+        value = 1;
+    }
```

Comment:
How is this a bug?
Also, would this match?
```suggestion
    // ! @ BUG: I believe multiple values got optimized out
    // value has to be a temp to match (v0) but there is only one value
    // explained by there being code where value got assigned with multiple conditions
    // suspicious: in vanilla there is only one cucco variant (0xFFFF)
    s32 value;
    if (1) {}
    value = 1;

```

Reply:
![if before ](https://user-images.githubusercontent.com/42048411/123584494-4f463700-d796-11eb-90bc-cb5f34a0d6b3.png)
No, it does not match.